src/openms/source/FORMAT/AbsoluteQuantitationMethodFile.cpp

```diff
@@ -38,176 +38,77 @@ namespace OpenMS
 {
   void AbsoluteQuantitationMethodFile::load(const String & filename, std::vector<AbsoluteQuantitationMethod> & aqm_list)
   {
-    // read in the .csv file
-    char is = ',';
-    bool ie = false; 
-    Int first_n = -1;
-    CsvFile::load(filename, is, ie, first_n);
-
-    // parse the file
-    std::map<String, Int> headers;
-    std::map<String, Int> params_headers;
-    StringList line, header;
-    AbsoluteQuantitationMethod aqm;
-    for (size_t i = 0; i < CsvFile::rowCount(); ++i)
-    {
-      if (i == 0) // header row
+    aqm_list.clear();
+    CsvFile::load(filename, ',', false, -1);
+    std::map<String, Size> headers;
+    StringList sl;
+    if (rowCount() >= 2) // no need to read headers if that's the only line inside the file
+    {
+      getRow(0, sl);
+      for (Size i = 0; i < sl.size(); ++i)
       {
-        CsvFile::getRow(i, header);
-        parseHeader_(header, headers, params_headers);
+        headers[sl[i]] = i; // for each header found, assign an index value to it
       }
-      else
+      if (!( // if any of these headers is missing, exception is thrown
```

Comment:
I am not sure if it is needed to throw an exception if a header is not found.  The code should provide a default value if the header is missing.  However, it might be useful for the reader to know if they missed a header if it was done by accident.

Reply:
Ok, so I'll give an empty string `""` to members of type `String` and `0` to numeric members. I'll look for a way to notify the user.